src/mixins/filing-template-mixin.ts

```diff
@@ -153,7 +149,13 @@ export default class FilingTemplateMixin extends DateMixin {
         correctedFilingDate: this.correctedFilingDate,
         comment: `${this.defaultCorrectionDetailComment}\n${this.getDetailComment}`,
         contactPoint: this.getContactPoint,
-        nameRequest: this.getNameRequest,
+        nameRequest: {
+          // add in the fields needed by Legal API
+          // don't save legal name if it's empty
+          ...this.getNameRequest,
+          legalName: this.getNameRequestLegalName || undefined,
+          nrNumber: this.getNameRequestNumber
+        },
```

Comment:
This is where the "conversion" from the store's NR object to the API's schema is done.

There is other code below to convert back from API to store formats.